pandas/core/series.py

```diff
@@ -330,7 +347,9 @@ def _init_dict(self, data, index=None, dtype=None):
             keys, values = [], []
 
         # Input is now list-like, so rely on "standard" construction:
-        s = Series(values, index=keys, dtype=dtype)
+        s = create_series_with_explicit_dtype(
+            values, index=keys, dtype=dtype, dtype_if_empty=np.float64
```

Comment:
Why do we want float64 for dictionary input?

Reply:
We don't really "want" it, but I wanted to replicate the current behaviour to ensure there's no unintended breaking change yet. This should obviously change when the default dtype of empty Series is really changed to `object`

Reply:
OK, can you add a TODO comment about that then?